File: 0601_0700/0667/0667.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m, k = (int(x) for x in f.readline().split())
    q.write(str((m + n) // k + ((m + n) % k > 0)) if k > 2 and m >= 2 * (n // (k - 2) + (n % (k - 2) > 0)) else '0')

File: 0601_0700/0667/0667.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, k;

	fscanf(f, "%d%d%d", &n, &m, &k);
	fprintf(q, "%d", k > 2 && m >= 2 * (n / (k - 2) + (n % (k - 2) > 0)) ? (m + n) / k + ((m + n) % k > 0) : 0);
	return 0;
}
